test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -951,3 +1009,70 @@ func getVMINameFromVMPod(vmPod *corev1.Pod) (string, error) {
 	}
 	return vmiRef.Name, nil
 }
+
+// returns a map of node --> list of kubevirtTestState objects for that node
+func getNodeVMTestStateMap(t *testing.T, ctxs []*scheduler.Context, allNodes map[string]node.Node) map[*node.Node][]*kubevirtTestState {
+	nodeTestStateMap := make(map[*node.Node][]*kubevirtTestState)
+	for _, appCtx := range ctxs {
+		testState := &kubevirtTestState{
+			appCtx:   appCtx,
+			allNodes: allNodes,
+		}
+		gatherInitialVMIInfo(t, testState)
+		verifyInitialVMI(t, testState)
+
+		// Add this testState to the node to VM map
+		vmNode, err := node.GetNodeByName(testState.vmPod.Spec.NodeName)
+		require.NoError(t, err, "failed to get node for %s: %v", testState.vmPod.Spec.NodeName)
+		nodeTestStateMap[&vmNode] = append(nodeTestStateMap[&vmNode], testState)
+	}
+	return nodeTestStateMap
+}
+
+func waitForVolumeAttachmentsToMove(t *testing.T, testState *kubevirtTestState) {
+	var err error
+	var oldAttachedNode, newAttachedNode *node.Node
+	var firstDisk *vmDisk
+
+	// we expect all volumes to be attached on the same node
```

Comment:
Before your changes, this check was done before restarting the volume driver. Now, it is being done after. That could cause this check to fail since the volume attachments will move at different times during sharedv4-svc failover.